src/api/contract/cluster-explorer/v1.ts

```diff
@@ -101,7 +101,7 @@ export namespace ClusterExplorerV1 {
     }
 
     export interface NodeSources {
-        resourceFolder(displayName: string, pluralDisplayName: string, manifestKind: string, abbreviation: string): NodeSource;
+        resourceFolder(displayName: string, pluralDisplayName: string, manifestKind: string, abbreviation: string, apiName: string): NodeSource;
```

Comment:
We can't do this, unfortunately - the `ClusterExplorerV1.NodeSources` interface is contractual, and this would break other extensions that depend on ours.  This is a tricky one.  We will have to either rev the ClusterExplorer API (and have a fallback for V1 consumers), or go back to your original strategy of inferring from the `pluralDisplayName`.  Neither of these solutions feels like much fun...

A _possible_ approach would be something like this:

* Make `apiName` an _optional_ member of ResourceKind, and _don't_ add it to the ClusterExplorerV1 API.  (We can add it to the ClusterExplorerV2 API when we release that.)
* In the `getTreeItem` implementations, look at whether the ResourceKind has an `apiName`.  If it _does_, append the string ` k8s-watchable` to the `contextValue`.
* In the `when` clauses in `package.json`, add ` && viewItem =~ /k8s-watchable/i`, so that only resources that have announced themselves as watchable will show the Watch and Stop Watching commands.

Again, I'm sorry for inflicting this extra fiddliness on you; I hadn't thought about the API compatibility implications when I suggested the ResourceKind approach.

Reply:
The above would perhaps be done indirectly by having `apiURI` return `undefined` if the ResourceKind didn't contain an `apiName`, and then appending ` k8s-watchable` only if `apiURI` was truthy.

Reply:
Done. I preferred to check the apiName. Feel free to change it if you want to use the apiURI

Reply:
I forgot to mention that our extensions (OpenShift Connector or Tekton extension which extend this one) adds new resources to the tree so we would need the ClusterExplorerV2 API to be released as well. Otherwise we will be unable to see the `Watch` action. Let me know if i can help to that side as well.